Add Katello plugin documentation

ehelms opened the pull request:
Opening up for comments on my approach for migrating over to be like other plugin documentation. There are still to dos for converting:
- [x] 2.4
- [x] 3.0
- [x] 3.1
- [x] 3.2
- [x] nightly

ehelms committed:
Add Katello plugin documentation

ehelms commented:
This is ready for some review. This captures all of the documentation for 2.4 to nightly with updated links and styling. The biggest thing not present is the API documentation. This adds a massive amount to the diff and a lot of files. Before we add it, I think it worth some re-thought how we share it. I noticed other plugins do not include their API documentation which seems worthy of generally solving.

stbenjam commented:
The site.version check isn't working, and all installation pages show:

 `These are the instructions for installing the unstable nightly release of katello!`

stbenjam commented:
If you click on the Install docs, and then look at the sidebar, all the links are unversioned as well.  It works from the initial page but not from the others

stbenjam commented:
A few other thoughts:
- How hard would it be to move these to the foreman manual style and get rid of our sidebar all together?
- Could we add some more information to the landing page? We can spruce up that /plugins/katello page a bit more than just the versions like other plugins have. Maybe at least say what the plugin does, and have our logo there
- But, otherwise it looks good!

ehelms commented:
On Thu, Oct 27, 2016 at 3:34 PM, Stephen Benjamin notifications@github.com
wrote:

> A few other thoughts:
> 
>    -
> 
>    How hard would it be to move these to the foreman manual style and get
>    rid of our sidebar all together?
> 
> Not hard, but I would ask what the benefit is. I'm down for whatever the
> majority of people think. I tend to find the single large page overwhelming.
> 
>    -
> 
>    Could we add some more information to the landing page? We can spruce
>    up that /plugins/katello page a bit more than just the versions like other
>    plugins have. Maybe at least say what the plugin does, and have our logo
>    there
> 
> Sure. I can work on a design and add it. Maybe take some stuff from the
> katello.org landing page as a starting place.
> 
>    -
> 
>    But, otherwise it looks good!
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/theforeman/theforeman.org/pull/726#issuecomment-256746359,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AAR5831IT3X612nhlmV0t8B9tO2I4lAjks5q4PzggaJpZM4KhpPW
> .

## 

Eric D. Helms
Red Hat Engineering
Ph.D. Student - North Carolina State University

ehelms commented:
Fixed the issue with the broken version links throughout and updated the highlighting of code blocks. I will tackle the landing page updates next.

ehelms commented:
@stbenjam pushed up a first pass at a landing page update

jlsherrill commented:
- Its unclear how to navigate to the nightly installation instructions, can we make that more clear?
- Nightly install doesn''t seem to render the  'required repositories' box upon page load.
- Nightly install's foreman yum localinstall url is incorrect (missing version)
- 2.4 install instructions is missing katello and foreman versions from their yum localinstall urls
- All client install instructions are missing the katello version from the katello-client-repos-latest.rpm url

ehelms commented:
@jlsherrill changed the following
- added nightly to list of versions on the right hand side of landing page
- updated nightly installation to show proper URLs and repo box by default
- updated 2.4 installation instructions
- updated client instructions for 3.2 and 2.4 which were the only two I saw broken

zjhuntin commented:
- I don't see a good way to return to the landing page upon picking a version's documentation.
- There are serveral pages in the user guide that currently will not load for me because they are missing index.md files :
- Content
- Content Views
- Docker Management
- lifecycle Environments
- Red Hat Content

Most of these seem easily fixed EG: I was able to get Content to load properly be changing the content.md file to index.md. Though I don't know if that's the solution we want to use.

ehelms commented:
@zjhuntin good find, I updated all versions of those to just use a consistent 'index.md'. For the returning to the landing page, that is a good point. I could change the top bar of each page to either:

```
<logo> Katello Documentation                           <version>
```

Where Katello is clickable and returns to landing page, or:

```
<logo Katello Documentation                            <version_dropdown>
```

Where versions can be navigated on any page and the Katello/logo is clickable to return to landing page.

ehelms commented:
Also, @stbenjam @jlsherrill if you have some opinions on the above comment

stbenjam commented:
I like the idea of a dropdown.

stbenjam commented:
But then again, where does the drop down link to? The other version of the current page? or the main page?  It might be harder to do the first if the page doesn't exist in that version.

ehelms commented:
@stbenjam I was thinking just going to main page for that documentation given the above. Thats how the current header dropdown on katello.org works.

stbenjam commented:
That's probably fine, not sure if users will find that confusing though.

zjhuntin commented:
I also like the idea of the drop down. Any way of making it a little faster to navigate through the area is going to make the site easier to use, I think.

ehelms commented:
@stbenjam added the dropdown

stbenjam commented:
@ehelms The dropdown shows up, but there's no entries in it for me.

Also looks like you committed a script called `change` into plugins/katello

ehelms commented:
@stbenjam what page(s) are you seeing empty dropdown?

stbenjam commented:
All of them

ehelms commented:
@stbenjam weird, it definitely works for me locally with all pages. What version of Jekyll?

stbenjam commented:
jekyll 3.1.6, let me bundle update and see what happens

stbenjam commented:
Still blank, and I get some warnings:

```
    Liquid Warning: Liquid syntax error (line 3): Unexpected character & in "page.version && page.version != 'nightly'" in plugins/katello/3.2/release_notes/release_notes.md
    Liquid Warning: Liquid syntax error (line 3): Unexpected character & in "page.version && page.version != 'nightly'" in plugins/katello/3.0/release_notes/release_notes.md
    Liquid Warning: Liquid syntax error (line 3): Unexpected character & in "page.version && page.version != 'nightly'" in plugins/katello/nightly/release_notes/release_notes.md
```

My bundle:

```
bundle show                                                                                                                                                                             add-katello 
Gems included by the bundle:
  * activesupport (4.2.7)
  * addressable (2.4.0)
  * bundler (1.10.5)
  * coffee-script (2.4.1)
  * coffee-script-source (1.10.0)
  * colorator (1.1.0)
  * ethon (0.9.1)
  * execjs (2.7.0)
  * faraday (0.9.2)
  * ffi (1.9.14)
  * forwardable-extended (2.6.0)
  * gemoji (2.1.0)
  * github-pages (102)
  * github-pages-health-check (1.2.0)
  * html-pipeline (2.4.2)
  * i18n (0.7.0)
  * icalendar (2.4.1)
  * jekyll (3.3.0)
  * jekyll-coffeescript (1.0.1)
  * jekyll-feed (0.7.2)
  * jekyll-gist (1.4.0)
  * jekyll-github-metadata (2.2.0)
  * jekyll-mentions (1.2.0)
  * jekyll-paginate (1.1.0)
  * jekyll-redirect-from (0.11.0)
  * jekyll-sass-converter (1.3.0)
  * jekyll-seo-tag (2.0.0)
  * jekyll-sitemap (0.11.0)
  * jekyll-swiss (0.4.0)
  * jekyll-watch (1.5.0)
  * jemoji (0.7.0)
  * json (1.8.3)
  * kramdown (1.11.1)
  * libv8 (3.16.14.15)
  * liquid (3.0.6)
  * listen (3.0.6)
  * mercenary (0.3.6)
  * mini_portile2 (2.1.0)
  * minima (2.0.0)
  * minitest (5.9.1)
  * multipart-post (2.0.0)
  * net-dns (0.8.0)
  * nokogiri (1.6.8.1)
  * octokit (4.6.0)
  * pathutil (0.14.0)
  * public_suffix (1.5.3)
  * rake (11.3.0)
  * rb-fsevent (0.9.8)
  * rb-inotify (0.9.7)
  * ref (2.0.0)
  * rouge (1.11.1)
  * safe_yaml (1.0.4)
  * sass (3.4.22)
  * sawyer (0.8.0)
  * terminal-table (1.7.3)
  * therubyracer (0.12.2)
  * thread_safe (0.3.5)
  * typhoeus (0.8.0)
  * tzinfo (1.2.2)
  * unicode-display_width (1.1.1)
```

ehelms commented:
@stbenjam that was it, newer jekyll didn't allow layouts to define page variables, instead I had to use layout. This should work now.

stbenjam commented:
Looking good.  Found a few things:
- Lots of 404's for images, e.g.: docker page (2.4), capsule (2.4), and content views (3.2), content hosts (3.1), etc
-  No side bar on some 2.4 pages: docker, lifecycle environments, red hat content, etc
-  API link 404 on all versions

ehelms commented:
All 404s should be fixed on images, as well as the missing sidebar. I have removed the side API link for now in favor of using `/apidoc` from a running server. I think the API display and generating needs work to only generate the plugin routes for display.

stbenjam commented:
- 3.1 content hosts still has 404's

```
[2016-11-10 15:22:00] ERROR `/list.png' not found.
[2016-11-10 15:22:00] ERROR `/details.png' not found.
[2016-11-10 15:22:00] ERROR `/content_host_content.png' not found.
[2016-11-10 15:22:00] ERROR `/subscriptions.png' not found.
[2016-11-10 15:22:00] ERROR `/packages.png' not found.
[2016-11-10 15:22:00] ERROR `/errata.png' not found.
[2016-11-10 15:22:00] ERROR `/select_all_results.png' not found.
[2016-11-10 15:22:00] ERROR `/host_collections.png' not found.
[2016-11-10 15:22:01] ERROR `/select_all_check.png' not found.
[2016-11-10 15:22:01] ERROR `/list.png' not found.
[2016-11-10 15:22:01] ERROR `/content_host_content.png' not found.
[2016-11-10 15:22:01] ERROR `/details.png' not found.
[2016-11-10 15:22:01] ERROR `/packages.png' not found.
[2016-11-10 15:22:01] ERROR `/errata.png' not found.
[2016-11-10 15:22:01] ERROR `/subscriptions.png' not found.
```

stbenjam commented:
Otherwise :+1:  Once you fix that, approved from me.  If anyone has any other feedback, should speak up now.

ehelms commented:
@zjhuntin @jlsherrill since you reviewed before, if you want to give any final input

ehelms commented:
Fixed the broken 3.1 content hosts page.

stbenjam commented:
Looks good to me, thanks @ehelms !
